[DeploymentScripts] inclusion of storage account settings

jorgecotillo opened the pull request:
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

jorgecotillo-fork committed:
inclusion of storage account settings

AutorestCI commented:
/azp run automation - sdk

azuresdkci commented:
Can one of the admins verify this patch?

jorgecotillo-fork committed:
included examples

AutorestCI commented:
/azp run automation - sdk

majastrz commented on specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/examples/DeploymentScripts_Create_using_existing_storage.json:
```diff
@@ -0,0 +1,111 @@
+          "lastModifiedByType": "Application",
+          "lastModifiedAt": "2020-02-01T01:01:01.1075056Z"
+        },
+        "properties": {
```
>properties [](start = 9, length = 10)

The response is missing the specified `storageAccountSettings`.

majastrz replied:
Also, what are we going to return in the PUT response if you don't specify `storageAccountSettings`?

---
In reply to: [412423575](https://github.com/Azure/azure-rest-api-specs/pull/9157#discussion_r412423575) [](ancestors = 412423575)

jorgecotillo replied:
@majastrz added storageAccountSettings to the response and also updated the sample from container group name. 
If storageAccountSettings are not provided, null will be returned as the value of storageAccountSettings.

Question, in a response that contains storageAccountSettings -> storageAccountName, storageAccountKey is set as an empty string because it is marked as a required property, does this results in the property being returned with an empty string value or it will be ignored from the response?

majastrz replied:
Properties that are secrets generally shouldn't be returned in the response or returned with explicit null (2nd case is rare). We shouldn't return empty string. The service can enforce that the storageAccountKey is present on PUT but it doesn't need to be returned in the response. I don't see any of these marked as required in the Swagger unless I'm missing something.

---
In reply to: [412492407](https://github.com/Azure/azure-rest-api-specs/pull/9157#discussion_r412492407) [](ancestors = 412492407)

majastrz reviewed (changes requested):
I added a comment. Please take a look.

jorgecotillo-fork committed:
removed restartPolicy

AutorestCI commented:
/azp run automation - sdk

jorgecotillo-fork committed:
updated sample response

AutorestCI commented:
/azp run automation - sdk

majastrz commented on specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json:
```diff
@@ -673,6 +684,20 @@
+          "type": "string",
+          "description": "The storage account name."
+        },
+        "storageAccountKey": {
```
>storageAccountKey [](start = 9, length = 17)

Can you add the `"x-ms-secret": true` to this property?

jorgecotillo commented:
Can someone tell me why renaming a file is throwing errors? The repo still looks for the previous file, since it does not exist, it throws errors.

raych1 commented:
/azp run public.rest-api-specs

raych1 commented:
@jorgecotillo , I just updated the version of tools. There's a reference file cannot be found. Please check the model validation result for details.

jorgecotillo-fork committed:
updated based on feedback

AutorestCI commented:
/azp run automation - sdk

jorgecotillo-fork committed:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

AutorestCI commented:
/azp run automation - sdk

jorgecotillo-fork committed:
renamed file

AutorestCI commented:
/azp run automation - sdk

jorgecotillo-fork committed:
updated file name to prevent case sensitive error

AutorestCI commented:
/azp run automation - sdk

jorgecotillo commented:
@raych1 thanks, the error was a case sensitive name update, I updated a file name to start with uppercase but the initial commit had the name with lowercase and git didn't catch the update, renamed the whole file and now is working.

majastrz reviewed (approved):
Signed off from ARM side.

raych1 commented:
Looks good to me, @jorgecotillo , let me know if this PR is ready to merge.

jorgecotillo commented on specification/resources/resource-manager/Microsoft.Resources/preview/2019-10-01-preview/deploymentScripts.json:
```diff
@@ -670,6 +675,32 @@
           "minLength": 1,
           "maxLength": 63
+        },
+        "restartPolicy": {
```
It is the same setting actually. In a recent PR update this property got removed (the RP got updated to not return this value, since it is read-only and sets only one value)

jorgecotillo-fork committed:
Merge branch 'master' of https://github.com/Azure/azure-rest-api-specs

AutorestCI commented:
/azp run automation - sdk

jorgecotillo commented:
@raych1 PR is ready to be merged. Thanks!
